packages/office-addin-mock/test/officeMockObject.test.ts

```diff
@@ -0,0 +1,153 @@
+
+import * as assert from "assert";
+import * as mocha from "mocha";
+import { OfficeMockObject } from "../src/main";
+
+const testObject = {
+  range: {
+    color: "blue",
+    getColor: function() {
+      return this.color;
+    },
+    font: {
+      size: 12,
+      type: "arial"
+    }
+  },
+}
+
+const contextMockData = {
+  workbook: {
+    range: {
+      address: "C2",
+    },
+    getSelectedRange: function () {
+      return this.range;
+    },
+  },
+};
+
+describe("Test OfficeMockObject class", function() {
+  describe("Populate object", function() {
+    it("Object structure created", async function() {
+      const officeMock = new OfficeMockObject(testObject);
+
+      officeMock.range.load("color");
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.color, "blue");
+
+      officeMock.range.font.load("size");
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.font.size, 12);
+
+      assert.strictEqual(officeMock.notAProperty, undefined);
+    });
+    it("Context mock created and working", async function() {
+      const contextMock = new OfficeMockObject(contextMockData);
+
+      const range = contextMock.workbook.getSelectedRange();
+      range.load("address");
+      await contextMock.sync();
+
+      assert.strictEqual(contextMock.workbook.range.address, "C2");
+    });
+    it("Multiple load calls", async function() {
+      const officeMock = new OfficeMockObject(testObject);
+
+      officeMock.range.load("color");
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.color, "blue");
+
+      officeMock.range.load("color");
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.color, "blue");
+
+      officeMock.range.load("color");
+      assert.strictEqual(officeMock.range.color, "blue");
+
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.color, "blue");
+
+      assert.strictEqual(officeMock.notAProperty, undefined);
+    });
+    it("Missing load", async function() {
+      const officeMock = new OfficeMockObject(testObject);
+      assert.strictEqual(officeMock.range.color, "Error, property was not loaded");
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.color, "Error, property was not loaded");
+    });
+    it("Missing sync", async function() {
+      const officeMock = new OfficeMockObject(testObject);
+      officeMock.range.load("color");
+      assert.strictEqual(officeMock.range.color, "Error, context.sync() was not called");
+      officeMock.sync();
+      officeMock.range.load("color");
+      assert.strictEqual(officeMock.range.color, "blue");
+    });
+    it("Functions added", async function() {
+      const officeMock = new OfficeMockObject(testObject);
+      officeMock.range.load("color");
+      officeMock.sync();
+      assert.strictEqual(officeMock.range.getColor(), "blue");
+      officeMock.range.color = "green";
+      assert.strictEqual(officeMock.range.getColor(), "green");
```

Comment:
I haven't tried it myself in the real Office.js code, but if I set a value does a sync need to happen for that to be committed?  I'm not sure that the correct code should be.

Reply:
You don't need to sync when setting the values. I just tested on the taskpane project.
So the current implementation is fine and working.